File: backend/src/Pokemon.Domain/Entities/Pokemon.cs
using FluentValidation.Results;
using Pokemon.Domain.Validators;

namespace Pokemon.Domain.Entities;

public class Pokemon: Entity
{
    public int PokemonTipoId { get; set; }
    public string Nome { get; set; } = null!;
    public string ImagemUrl { get; set; } = null!;
    public string Descricao { get; set; } = null!;
    public PokemonTipo PokemonTipo { get; set; } = null!;
    
    public override bool Validar(out ValidationResult validationResult)
    {
        validationResult = new PokemonValidator().Validate(this);
        return validationResult.IsValid;
    }
}

File: frontend/angular/src/app/shared/domain-types/models/pokemon.ts
import { PokemonTipoViewModel } from ".";

export interface PokemonViewModel {
  id: number;
  nome: string;
  imagemUrl: string;
  descricao: string;
  pokemonTipo: PokemonTipoViewModel;
}

export interface AddPokemonInputModel {
  pokemonTipoId: number;
  nome: string;
  imagemUrl: string;
  descricao: string;
}

export interface AddPokemonViewModel {
  id: number;
  pokemonTipoId: number;
  nome: string;
  imagemUrl: string;
  descricao: string;
}

export interface UpdatePokemonInputModel {
  id: number;
  pokemonTipoId: number;
  nome: string;
  imagemUrl: string;
  descricao: string;
}

export interface UpdatePokemonViewModel {
  pokemonTipoId: number;
  nome: string;
  imagemUrl: string;
  descricao: string;
}